src/createFilter.ts

```diff
@@ -1,13 +1,21 @@
-import * as mm from 'micromatch';
+import mm from 'micromatch';
 import { resolve, sep } from 'path';
 import { CreateFilter } from './pluginutils';
 import ensureArray from './utils/ensureArray';
 
+function sanitizeResolutionBase(resolutionBase: string): string {
+	return resolutionBase.replace(/[()]/g, '\\$&');
+}
+
 function getMatcherString(id: string, resolutionBase: string | false | null | undefined) {
-	if (resolutionBase === false) {
-		return id;
-	}
-	return resolve(...(typeof resolutionBase === 'string' ? [resolutionBase, id] : [id]));
+	return resolutionBase === false
+		? id
+		: resolve(
+				sanitizeResolutionBase(
```

Comment:
myb putting the sanitation after the resolve would work

Reply:
Good thinking. Problem is we want to sanitize the resolution base but not the id, as it needs to support micromatch patterns. Let's see if I can think of something.

Reply:
No, that does not seem to be it. I'm really at a loss here.

Reply:
what can the id be? as in what actually gets resolved here

Reply:
id is probably badly named. It is any micromatch pattern the user provides.

Reply:
E.g. `src/**/*.js` should be resolved to something like `/Users/moi/project/src/**/*.js`

Reply:
ah ok, wasn't sure resolve would deal with wildcards like that